neqo-transport/src/connection/mod.rs

```diff
@@ -2491,6 +2530,18 @@ impl Connection {
                 self.acks
                     .ack_freq(seqno, tolerance - 1, delay, ignore_order);
             }
+            Frame::Datagram { data, .. } => {
+                let max_size = self
+                    .tps
+                    .borrow()
+                    .local
+                    .get_integer(tparams::MAX_DATAGRAM_FRAME_SIZE);
```

Comment:
We should be able to get this from the connection parameters instead.  That's immutable, so it won't be different.  And it doesn't need this whole borrow/lookup thing.

You should be able to copy the value to a new object if you move this processing elsewhere, but this check could just stay here.